File: 680.验证回文字符串-ⅱ.java
/*
 * @lc app=leetcode.cn id=680 lang=java
 *
 * [680] 验证回文字符串 Ⅱ
 */

// @lc code=start
class Solution {
    private char[] arr;
    public boolean validPalindrome(String s) {
        this.arr = s.toCharArray();
        int start = 0;
        int end = s.length() - 1;
        while (start <= end) {
            if (arr[start] == arr[end]) {
                start++;
                end--;
            } else {
                return helper(start, end-1) || helper(start+1, end);
            }
        }
        return true;
    }

    private boolean helper(int start, int end) {
        while (start <= end) {
            if (arr[start] == arr[end]){
                start++;
                end--;
            } else {
                return false;
            }
        }
        return true;
    }
}
// @lc code=end

File: 680.验证回文字符串-ⅱ.py
#
# @lc app=leetcode.cn id=680 lang=python3
#
# [680] 验证回文字符串 Ⅱ
#

# @lc code=start


class Solution:
    def validPalindrome(self, s: str) -> bool:
        start = 0
        end = len(s) - 1
        while start < end:
            if s[start] == s[end]:
                start += 1
                end -= 1
            else:
                print(start)
                return s[start + 1:end + 1] == s[end:start:-1] or\
                    s[start:end] == ''.join(reversed(s[start:end]))
        return True

# @lc code=end
